iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveHvdcLine.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.AbstractNetworkModification;
+import com.powsybl.iidm.network.HvdcConverterStation;
+import com.powsybl.iidm.network.HvdcLine;
+import com.powsybl.iidm.network.Network;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.powsybl.iidm.modification.topology.ModificationReports.*;
+
+/**
+ * @author Anis Touri <anis-1.touri@rte-france.com>
+ */
+public class RemoveHvdcLine extends AbstractNetworkModification {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoveHvdcLine.class);
+
+    private final String hvdcLineId;
+    private final List<String> mscIds;
+
+    public RemoveHvdcLine(String hvdcLineId, List<String> mscIds) {
+        this.hvdcLineId = Objects.requireNonNull(hvdcLineId);
+        this.mscIds = mscIds;
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, ComputationManager computationManager, Reporter reporter) {
+        AtomicBoolean hvdcLineRemoved = new AtomicBoolean(false);
```

Comment:
What is the use of atomic values there, how it could synchronize/guaranty anything, as read and writes seem to be on same thread ?